modules/core/src/transitions/gpu-spring-transition.js

```diff
@@ -0,0 +1,240 @@
+/* eslint-disable complexity, max-statements, max-params */
+import GL from '@luma.gl/constants';
+import {Buffer, Transform} from '@luma.gl/core';
+import {padBuffer} from '../lib/attribute-transition-utils';
+import BaseAttribute from '../lib/base-attribute';
+import Attribute from '../lib/attribute';
+
+export default class GPUSpringTransition {
+  constructor({gl, attribute, transitionSettings}) {
+    this.type = 'spring';
+    this.attribute = attribute;
+    this.transitionSettings = transitionSettings;
+    // this is the attribute we return during the transition
+    // `attribute.userData` is the original options passed when constructing the attribute.
+    // This ensures that we set the proper `doublePrecision` flag and shader attributes.
+    this.attributeInTransition = new Attribute(gl, attribute.userData);
+    this.attributeInTransition.update({
```

Comment:
`constant` will be set to `false` when you update this attribute with a buffer.

Reply:
I mostly have this line here to show the reader the intention of this code. Additionally, it seems bad to rely on what feels like an implementation detail of `update` to make sure `constant` is set to `false`. Thoughts?

Reply:
It is by design that `constant` cannot be `true` if a buffer is present. You should never use this attribute instance without updating it anyways.

Reply:
I can remove this code, but I want to add a comment. It is not obvious at all to a user of the `AttributeTransitionManager` that if you pass a constant attribute, it will be converted to a regular attribute. I want to make sure it is clear to the reader that this is the intended contract of our API.